bytecode/instr.py

```diff
@@ -144,6 +145,37 @@ def _check_arg_int(name, arg):
                          % name)
 
 
+_stack_effects = {
+    # NOTE: the entries are all 3-tuples, to leverage tuple[-1] being the last
+    # value.  Entry[0] is non-taken jumps.  Entry[1] is for taken jumps.
+    # Entry[2] is the max of the others.  It is maintained manually here but
+    # verified in a unittest.  Entry [2] is so that we do not need a runtime
+    # test when callers of stack_effect() specify "jump=-1".  Remember that
+    # Entry[2] is also addressable as Entry[-1].
+
+    # opcodes not in dis.stack_effect
+    _opcode.opmap['EXTENDED_ARG']: (0, 0, 0),
+    _opcode.opmap['NOP']: (0, 0, 0),
+
+    # Jump taken/not-taken are different:
+    _opcode.opmap['JUMP_IF_TRUE_OR_POP']: (-1, 0, 0),
+    _opcode.opmap['JUMP_IF_FALSE_OR_POP']: (-1, 0, 0),
+    _opcode.opmap['FOR_ITER']: (1, -1, 1),
+    _opcode.opmap['SETUP_WITH']: (1, 6, 6),
+    _opcode.opmap['SETUP_ASYNC_WITH']: (0, 5, 5),
+    _opcode.opmap['SETUP_EXCEPT']: (0, 6, 6),   # as of 3.7, below for <=3.6
+    _opcode.opmap['SETUP_FINALLY']: (0, 6, 6),  # as of 3.7, below for <=3.6
+}
+
+# More stack effect values that are unique to the version of Python.
+if not (sys.version_info >= (3, 7)):
```

Comment:
Why not just `sys.version_info < (3, 7)`?

Reply:
I originally wrote it as "sys.version_info <= (3,6)", which was the problem.  Yes, apparently "<" works but "<=" does not.  Although I would prefer "<= (3,6)", it is certainly cleaner to say "< (3,7)" than "not ( >= (3,7))"

```Python 3.7.0b3 (v3.7.0b3:4e7efa9c6f, Mar 29 2018, 08:11:44) 
[Clang 6.0 (clang-600.0.57)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import sys
>>> sys.version_info
sys.version_info(major=3, minor=7, micro=0, releaselevel='beta', serial=3)
>>> sys.version_info < (3,8)
True
>>> sys.version_info < (3,7)
False
>>> sys.version_info <= (3,7)
False
```

Ahhh, I see.  That was a noob mistake.  Apparently "<= (3,7)" is short for "<= (3,7,0,0,0)" (or something like that) and I'm running (3,7,0,'beta',3) which is greater.  So I've just convinced myself why my preference for "<= (3,6)" is just wrong.  Thanks for making me think that through.